engine/allocator.hpp

```diff
@@ -192,8 +180,7 @@ class Allocator {
   uint64_t max_offset_;
 
   bool thread_local_counter_enabled_{};
-  uint32_t num_thread_local_counters{0};
-  std::int64_t* thread_allocated_sizes_{nullptr};
+  std::vector<int64_t> thread_allocated_sizes_{};
```

Comment:
`std::vector` is supposed to have potential bottleneck here. This was discussed before : https://github.com/pmem/kvdk/pull/329#pullrequestreview-1099977968

Reply:
I think std::int64* doesn't resolve this problem, we need to align it to cache line

Reply:
OK. I misunderstood this problem.

Reply:
Will you do the alignment in this pull request?

Reply:
yes